code/clue.js

```diff
@@ -1,59 +1,464 @@
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  mystery.favouriteWeapon = mystery.killer.favouriteWeapon;
+  killerLoader.style.display = "none";
+  killerCard.style.background = mystery.killer.color;
+  killerName.innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  killerAge.innerHTML = `Age: ${mystery.killer.age}`;
+  killerFavouriteWeapon.innerHTML = `Favourite weapon: ${mystery.killer.favouriteWeapon}`;
+  killerImage.style.display = "inline-block";
+  killerImage.src = mystery.killer.image;
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+killerCard.onclick = killerAnimation;
+
+// Actions related to Weapon deck
+
+const favouriteWeaponAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else {
+    weaponLoader.style.display = "block";
+    weaponName.innerHTML = '';
+    weaponWeight.innerHTML = '';
+    weaponImage.src = '';
+    weaponImage.style.display = "none";
+    weaponFavouriteRoom.innerHTML = '';
+    revealedMystery.innerHTML = '';
+    setTimeout(function () {
+      pickFavouriteWeapon(mystery.killer);
+    }, 2000);
+  };
+};
+
+const favouriteWeaponDuplicated = (elementToPopulate, howManyTimes) => {
+  let result = [];
+  for (let i = 0; i < howManyTimes; i++) {
+    result.push(elementToPopulate);
+  };
+  return result;
+};
+
+const pickFavouriteWeapon = (suspect) => {
+  const newArrayOfWeapons = weapons.map(a => a.name);
+  const randomWeapons = [
+    ...newArrayOfWeapons,
+    ...favouriteWeaponDuplicated(suspect.favouriteWeapon, 10)
+  ];
+  mystery.pickedWeapon = randomSelector(randomWeapons);
+  const mysteryWeapon = weapons.find((item) => {
+    if (item.name === mystery.pickedWeapon) {
+      return item;
+    };
+  });
+  mystery.weapon = mysteryWeapon;
+  mystery.favouriteRoom = mysteryWeapon.favouriteRoom;
+  weaponLoader.style.display = "none";
+  weaponCard.style.background = "#808080";
+  weaponName.innerHTML = mystery.weapon.name;
+  weaponWeight.innerHTML = `Weight: ${mystery.weapon.weight} oz`;
+  weaponImage.style.display = "inline-block";
+  weaponImage.src = mystery.weapon.image;
+  weaponFavouriteRoom.innerHTML = `Location: ${mystery.weapon.favouriteRoom}`;
+};
+
+weaponCard.onclick = favouriteWeaponAnimation;
+
+// Actions related to Room deck
+
+const favouriteRoomAnimation = () => {
+  if (mystery.killer === undefined) {
+    revealedMystery.innerHTML = ('Pick a killer first!');
+  } else if (mystery.weapon === undefined) {
+    revealedMystery.innerHTML = ('Pick a weapon first!');
+  } else {
```

Comment:
Awesome job with the user prompts! 👏🏻

Reply:
This could be written like:
`if (!mystery.killer)`